src/bootstrap/config.rs

```diff
@@ -1421,8 +1496,18 @@ impl Config {
             if let Some(true) = rust.incremental {
                 config.incremental = true;
             }
-            set(&mut config.use_lld, rust.use_lld);
+            set(&mut config.lld_mode, rust.lld_mode);
             set(&mut config.lld_enabled, rust.lld);
+
+            if matches!(config.lld_mode, LldMode::SelfContained)
+                && !config.lld_enabled
+                && flags.stage.unwrap_or(0) > 0
```

Comment:
Since switching to MCP510, we should be using the self-contained `lld` coming from the compiler that is actually being used for compiling, and I think that makes sense. I'm not sure if it would be that easy with MCP510 to tell `rustc` to use self-contained `lld`, but use `rust-lld` from a different sysroot for that. So the approach from this PR makes sense to me.

Reply:
Hmm, I just realized that I skipped this check after rebasing. But when I tried it locally with
```toml
lld = false
use-lld = "self-contained"
```
I can do a stage2 build just fine. That's weird, I don't know where is the `lld` coming from.

Reply:
Oh, it looks like it still uses the global `lld` if a self-contained one wasn't found. I guess this is the expected behaviour of MCP510. So just adding the check back should probably be enough.